src/coreComponents/physicsSolvers/multiphysics/MultiphasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,554 @@
+    {
+      GEOSX_ERROR( "MultiphasePoroelastic solver allows at most three components at the moment" );
+    }
+  }
+
+  //*****************************************************************************
+  /**
+   * @class StackVariables
+   * @copydoc geosx::finiteElement::ImplicitKernelBase::StackVariables
+   *
+   * Adds a stack array for the displacement, incremental displacement, and the
+   * constitutive stiffness.
+   */
+  struct StackVariables : public Base::StackVariables
+  {
+public:
+
+    static constexpr int numDispDofPerElem =  Base::StackVariables::numRows;
+
+    /// Constructor.
+    GEOSX_HOST_DEVICE
+    StackVariables():
+      Base::StackVariables(),
+            xLocal(),
+            u_local(),
+            uhat_local(),
+            localFlowResidual{ 0.0 },
+      localDispFlowJacobian{ {0.0} },
+      localFlowDispJacobian{ {0.0} },
+      localFlowFlowJacobian{ {0.0} },
+      localFlowDofIndex{ 0 }
+    {}
+
+#if !defined(CALC_FEM_SHAPE_IN_KERNEL)
+    /// Dummy
+    int xLocal;
+#else
+    /// C-array stack storage for element local the nodal positions.
+    real64 xLocal[ numNodesPerElem ][ 3 ];
+#endif
+
+    /// Stack storage for the element local nodal displacement
+    real64 u_local[numNodesPerElem][numDofPerTrialSupportPoint];
+
+    /// Stack storage for the element local nodal incremental displacement
+    real64 uhat_local[numNodesPerElem][numDofPerTrialSupportPoint];
+
+    real64 localFlowResidual[numMaxComponentsMultiphasePoroelastic];
+    real64 localDispFlowJacobian[numDispDofPerElem][numMaxComponentsMultiphasePoroelastic + 1];
+    real64 localFlowDispJacobian[numMaxComponentsMultiphasePoroelastic][numDispDofPerElem];
+    real64 localFlowFlowJacobian[numMaxComponentsMultiphasePoroelastic][numMaxComponentsMultiphasePoroelastic + 1];
+
+    /// C-array storage for the element local row degrees of freedom.
+    globalIndex localFlowDofIndex[numMaxComponentsMultiphasePoroelastic+1];
+
+  };
+  //*****************************************************************************
+
+  /**
+   * @brief Copy global values from primary field to a local stack array.
+   * @copydoc ::geosx::finiteElement::ImplicitKernelBase::setup
+   *
+   * For the SinglePhase implementation, global values from the displacement,
+   * incremental displacement, and degree of freedom numbers are placed into
+   * element local stack storage.
+   */
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void setup( localIndex const k,
+              StackVariables & stack ) const
+  {
+    for( localIndex a=0; a<numNodesPerElem; ++a )
+    {
+      localIndex const localNodeIndex = m_elemsToNodes( k, a );
+
+      for( int i=0; i<3; ++i )
+      {
+#if defined(CALC_FEM_SHAPE_IN_KERNEL)
+        stack.xLocal[ a ][ i ] = m_X[ localNodeIndex ][ i ];
+#endif
+        stack.u_local[ a ][i] = m_disp[ localNodeIndex ][i];
+        stack.uhat_local[ a ][i] = m_uhat[ localNodeIndex ][i];
```

Comment:
I wonder if we should use `disp` and `deltaDisp` like we do for the pressure instead of `u` and `u_hat`.

Reply:
I agree that the naming convention should be improved. However, it should reflected across all solid mechanics kernels.